### PR TITLE
use direct field access in A_mul_B_td!

### PR DESCRIPTION
**Pull request by fredrikekre**
Should fix the regression of matvec and matmat multiplication with special matrices (see #24577). I accidentally introduced this in #24324

@nanosoldier `runbenchmarks("linalg", vs = ":master")`

### CODE REVIEW

**Review by strickek — COMMENTED** [review]
check line 367 - OR with the same on both sides.

**Comment by fredrikekre** [comment]
@nanosoldier `runbenchmarks("linalg", vs = ":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/24578#issuecomment-343843349) has completed - no performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/ae05c5b795f798f085616192f17133e50f28b3d8/cfb701d_vs_daa58bb/report.md). cc @ararslan

File: base/linalg/bidiag.jl [hunk 1]
```diff
@@ -358,17 +358,30 @@ function check_A_mul_B!_sizes(C, A, B)
     end
 end
 
+# function to get the internal vectors for Bidiagonal and [Sym]Tridiagonal
```
Comment (andreasnoack) [inline base/linalg/bidiag.jl]: Maybe add that this is useful because it avoids the allocation when the diagonal exists

**Pull request merged**